### PR TITLE
08 arrays

### PR DESCRIPTION
**Pull request by riraum**

### CODE REVIEW

File: 08_arrays/main.go [hunk 1]
```diff
@@ -43,40 +43,173 @@ func main() {
...
+
+	// for {
+	// 	count4 > 9 {
+	// 		break
+	// 	}
+	// 	fillerMultiplier = intArray[count4] * filler
+	// 	fmt.Println(fillerMultiplier)
+	// 	count4++
+	// }
+
```
Comment (riraum) [inline 08_arrays/main.go]: @nobe4 Bit stuck here, any hints?
I'm thinking I might have to loop twice and use append somehow, to get the "x" into the array. But how to put everything together, I don't get right now.

Comment (nobe4) [reply 08_arrays/main.go]: Try to divide the task in 2 smaller tasks.

What would be the first and second task?

Comment (riraum) [reply 08_arrays/main.go]: Loop through `intArray` to concatenate `x` in a loop as often as the respective element.
Add/append the result to the new array `lengthIntArray`.

Comment (nobe4) [reply 08_arrays/main.go]: Yeah, I would have put it slightly differently but you have it essentially:

1. build a string by concatenating `x` multiple times together
2. loop through the array to create a new one with the result of `1.`

`1.` can be made into a function (maybe you've even already done something similar?)

Comment (riraum) [reply 08_arrays/main.go]: I think I have both 1. and 2.
I'm now facing the issue again that I can't declare a function inside the another/the main function.
I tried closing the main function and moving the function and the required parts outside with it and then calling it in the loop.
Seems a bit harder compared to `06_functions` where I could just call the function inside the main function.
I do have to move the function outside and close `main` first, no?

Comment (nobe4) [reply 08_arrays/main.go]: yes, you can't define a function inside another one. Move it out of the main function.

File: 08_arrays/main.go [hunk 2]
```diff
@@ -43,45 +43,213 @@ func main() {
...
+	stringArray[8] = "A"
+	stringArray[9] = "POTATO"
+	fmt.Println(stringArray)
+	fmt.Println("len test=", len(stringArray[0]))
+
+	var lengthStringArray [10]int
+	var count3 = 0
+
+	for {
+		if count3 > 9 {
```
Comment (nobe4) [inline 08_arrays/main.go]: Use the length of the array here.

File: 08_arrays/main.go [hunk 3]
```diff
@@ -43,45 +43,213 @@ func main() {
...
+
+	// create new empty array of 10 strings
+	var lengthIntArray []string
+	// counter
+	var arrayAssemblyCounter = 0
+
+	// loop to create output array
+	for {
+		// if counter is > than the last/[10] array element break the loop
+		if arrayAssemblyCounter > intArray[arrayAssemblyCounter]-1 {
```
Comment (nobe4) [inline 08_arrays/main.go]: This condition can be simplified.

File: 08_arrays/main.go [hunk 4]
```diff
@@ -43,45 +43,213 @@ func main() {
...
+// 	return arrayElementOne + arrayElementTwo
+// }
+
+// intArray function below
+// create empty variable for "x" string
+var arrayString string = ""
+
+// counter
+var concatCounter = 0
+
```
Comment (nobe4) [inline 08_arrays/main.go]: Those variables need to be in the function.

File: 08_arrays/main.go [hunk 5]
```diff
@@ -43,45 +43,231 @@ func main() {
...
+	var count int = 1
+	var times int = 10
+
+	for {
+		if count > times {
+			break
+		}
+		tenInt = append(tenInt, count)
+		count++
+	}
```
Comment (nobe4) [inline 08_arrays/main.go]: Here you can use a single variable for the counter.

File: 08_arrays/main.go [hunk 6]
```diff
@@ -43,45 +43,231 @@ func main() {
...
+	fmt.Println(tenInt[len(tenInt)-1])
 
 	// TODO: Using the array from the previous step, create a new array
 	// containing the first 5 elements multiplied by 2.
 	// Hint: Use a loop.
 	// Hint: The new array should contain [2, 4, 6, 8, 10].
 	// Print the new array.
 
+	// create copy of array
+	// loop 5 times (first 4 elements needed)
```
Comment (nobe4) [inline 08_arrays/main.go]:
```suggestion
	// loop 5 times (first 5 elements needed)
```

File: 08_arrays/main.go (hunk 4 above)
Comment (riraum) [reply 08_arrays/main.go]: It did seem to run like this, but moved them!

File: 08_arrays/main.go [hunk 7]
```diff
@@ -43,45 +43,230 @@ func main() {
...
+	var multCounter = 0
+	for {
+		// TODO: replace hard coded value with last index of
+		if multCounter > 9 {
+			break
+		}
+		multArray = append(multArray, anotherIntArray[multCounter]*yetAnotherIntArray[multCounter])
+		// increase counter
+		multCounter++
+	}
```
Comment (nobe4) [inline 08_arrays/main.go]: make this with just one loop, the extra 2 are redundant

Comment (riraum) [reply 08_arrays/main.go]: Done, good one. Felt a bit dirty writing it, thought of variables, but this is better.

File: 08_arrays/main.go (hunk 6 above)
Comment (riraum) [reply 08_arrays/main.go]: Done. Mixed it up with the element position of the array.

File: 08_arrays/main.go (hunk 5 above)
Comment (riraum) [reply 08_arrays/main.go]: Done.

File: 08_arrays/main.go (hunk 2 above)
Comment (riraum) [reply 08_arrays/main.go]: Yeah that bugged me, had more like this and was on my TODO list. Done.

File: 08_arrays/main.go (hunk 3 above)
Comment (riraum) [reply 08_arrays/main.go]: Of course, done.

**Review by nobe4 — APPROVED** [review]

**Pull request merged**